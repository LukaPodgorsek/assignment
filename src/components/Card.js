import React, { Component } from "react";

// Grommet
import { Box } from "grommet";

class Card extends Component {
  render() {
    const { children, ...rest } = this.props;
    return (
      <Box
        margin={{ vertical: "small" }}
        elevation="medium"
        background="background-front"
        {...rest}
      >
        {children}
      </Box>
    );
  }
}

Card.propTypes = {};

export default Card;
