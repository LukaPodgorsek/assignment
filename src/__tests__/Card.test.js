import { render } from "@testing-library/react";
import Card from "../components/Card";

test("renders plain Card", () => {
  const { container } = render(<Card />);
  expect(container.firstChild).toMatchSnapshot();
});

test("renders Card with props", () => {
  const { container } = render(<Card background="white" />);
  expect(container.firstChild).toMatchSnapshot();
});
