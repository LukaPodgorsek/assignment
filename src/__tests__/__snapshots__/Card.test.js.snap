// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card with props 1`] = `
<div
  class="StyledBox-sc-13pk1d4-0 vIbJK"
/>
`;

exports[`renders plain Card 1`] = `
<div
  class="StyledBox-sc-13pk1d4-0 vIbJK"
/>
`;
